### DISCUSSION TITLE
Feature request: Startup parameter to enable/disable mapstruct

### DISCUSSION THREAD

**Discussion by RichMacDonald** [mapstruct/mapstruct#3750] [Ideas]
I use maven and the eclipse compiler. My parent pom.xml contains the maven-compiler-plugin configuration. Mapstruct is included as an annotation. However, only a few of my sub-projects use mapstruct. I could speed up compilation by turning off mapstruct for those projects that don't use it.

I would like to enable mapstruct for some projects and disable it for others using the pom.xml. I have not found a way to accomplish this with maven profiles. I could create two intermediate maven modules (one with mapstruct and one without) but that is awkward.

The most convenient way might be to include mapstruct as a processor in the single declaration but turn it on and off using a startup parameter and a maven variable:

````<arg>-Amapstruct.enable=true/false</arg>````

In maven, this can then be rewritten as:

````
<mapstruct.enabled>true</mapstruct.enabled>
<arg>${mapstruct.enabled}</arg>`
````

Is this a reasonable idea?

**Comment by filiphr** [comment]
This looks more like a maven / gradle feature rather than a MapStruct one. What kind of a performance impact are we talking about? Have you measured? Can you perhaps run a profiler compiling with and without MapStruct and share that with me?

Keep in mind that even if we did add a parameter that we would check for whether to run the processor or not, I seriously doubt that it will improve the performance. There could be other things that we might be able to do to improve the performance on an empty set of annotations.
